src/ui/public/doc_table/components/table_row.js

```diff
@@ -92,18 +102,30 @@ module.directive('kbnTableRow', function ($compile) {
           openRowHtml
         ];
 
+        const mapping = indexPattern.fields.byName;
         if (indexPattern.timeFieldName) {
           newHtmls.push(cellTemplate({
             timefield: true,
-            formatted: _displayField(row, indexPattern.timeFieldName)
+            formatted: _displayField(row, indexPattern.timeFieldName),
+            filterable: mapping[indexPattern.timeFieldName].filterable,
+            column: indexPattern.timeFieldName
           }));
         }
 
         $scope.columns.forEach(function (column) {
+          const flattenedRow = indexPattern.flattenHit(row);
+          const isFilterable =
+            flattenedRow[column] === undefined
+            ? false
+            : !mapping[column]
+            ? false
+            : mapping[column].filterable;
           newHtmls.push(cellTemplate({
             timefield: false,
             sourcefield: (column === '_source'),
-            formatted: _displayField(row, column, true)
+            formatted: _displayField(row, column, true),
```

Comment:
@Bargs I was worried the raw field value could be quite large and contain special characters, quotes etc - would this be safe to pass into an HTML element attribute and back into inlineFilter unscathed?